packages/near-membrane-base/src/membrane.ts

```diff
@@ -626,93 +635,17 @@ export function createMembraneMarshall() {
-                            writable,
-                            valuePointer,
-                            getPointer,
-                            setPointer
-                        ) => {
-                            ReflectDefineProperty(
-                                shadowTarget,
-                                key,
-                                createDescriptorFromMeta(
-                                    configurable,
-                                    enumerable,
-                                    writable,
-                                    valuePointer,
-                                    getPointer,
-                                    setPointer
-                                )
-                            );
-                        }
-                    );
-                }
-            }
-            return result;
-        }
-
-        function liveDeleteProperty(
-            foreignTargetPointer: Pointer,
-            _shadowTarget: ShadowTarget,
-            key: string | symbol
-        ): ReturnType<typeof Reflect.deleteProperty> {
-            return foreignCallableDeleteProperty(foreignTargetPointer, key);
-        }
-
-        function liveGet(
-            foreignTargetPointer: Pointer,
-            shadowTarget: ShadowTarget,
-            key: string | symbol,
-            receiver: any
-        ): any {
-            const safeDesc =
-                liveGetOwnPropertyDescriptor(foreignTargetPointer, shadowTarget, key) ||
-                getInheritedPropertyDescriptor(foreignTargetPointer, key);
-
-            if (safeDesc) {
-                const { get: getter, value: localValue } = safeDesc;
-                if (getter) {
-                    // Even though the getter function exists, we can't use
-                    // `ReflectGet` because there might be a distortion for
-                    // that getter function, in which case we must resolve
-                    // the local getter and call it instead.
-                    return ReflectApply(getter, receiver, []);
-                }
-                return localValue;
-            }
-            if (key === TO_STRING_TAG_SYMBOL) {
-                return foreignCallableGetUnbrandedTag(foreignTargetPointer);
-            }
-            return undefined;
+        function lockShadowTarget(shadowTarget: ShadowTarget, foreignTargetPointer: Pointer) {
+            // set prototype after to avoid
+            copyForeignDescriptorsIntoShadowTarget(shadowTarget, foreignTargetPointer);
+            // setting up __proto__ of the shadowTarget
+            const proto = getLocalValue(foreignCallableGetPrototypeOf(foreignTargetPointer));
+            ReflectSetPrototypeOf(shadowTarget, proto);
+            // locking down the extensibility of shadowTarget
+            ReflectPreventExtensions(shadowTarget);
         }
 
-        function liveGetOwnPropertyDescriptor(
+        function foreignGetOwnPropertyDescriptor(
```

Comment:
hmm, this sounds suspicious. In the original design, something prefixed with `foreign` cannot be declared, it can only come into play via a callable argument when the two sides of the membrane are connected. This breaks that invariant, and now the confusion is whether or not this is coming from the other side.

Reply:
It's not breaking the invariant it's just a confusing name. I'll noodle on another name for it.

Reply:
Renamed the methods to avoid the "foreign" name.